projects/packages/sync/src/sync-queue/class-queue-storage-table.php

```diff
@@ -456,4 +476,190 @@ public function delete_items_by_ids( $ids ) {
 			)
 		);
 	}
+
+	/**
+	 * Table initialization
+	 */
+	public static function initialize_custom_sync_table() {
```

Comment:
Would it make sense to rename this to `enable_custom_sync_table` and include the migration logic here? Because `migrate_from_options_table_to_custom_table` is currently public but it should always run after ensuring that the table exists and is healthy.

Reply:
It is public so it can be ran separately from external source, as it's used in #32275